### PR TITLE
nil pointer fixes

### PR DESCRIPTION
**Pull request by GregHanson**
**Please provide a description of this PR:**

a few nil pointer checks

### CODE REVIEW

File: operator/pkg/validate/validate.go [hunk 1]
```diff
@@ -222,7 +222,9 @@ func validateRevision(_ util.Path, val any) util.Errors {
 func validateGatewayName(path util.Path, val any) (errs util.Errors) {
 	v := val.([]*v1alpha1.GatewaySpec)
 	for _, n := range v {
-		errs = append(errs, validateWithRegex(path, n.Name, ObjectNameRegexp)...)
+		if n != nil {
```
Comment (howardjohn) [inline operator/pkg/validate/validate.go:225]: Should probably reject if it is nil

Comment (howardjohn) [reply operator/pkg/validate/validate.go:225]: unless that is somehow valid?

File: pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go [hunk 2]
```diff
@@ -136,7 +136,9 @@ func applyLocalityWeight(
 
 			// remove groups of endpoints in a locality that miss matched
 			for i := range misMatched {
-				loadAssignment.Endpoints[i].LbEndpoints = nil
+				if loadAssignment.Endpoints[i] != nil {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go:139]: This seems suspicious. How does this happen? is it from a test hitting only low level functions with obscure inputs or can it really be triggered?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
lgtm but hold since I had some questions

File: pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go (hunk 2 above)
Comment (GregHanson) [reply pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go:139]: This one I wasn't too sure of - but I think the culprit is `FuzzApplyLocalityLBSetting` in `pilot/pkg/networking/core/v1alpha3/loadbalancer/fuzz_test.go`.  So I'm assuming that falls under the category of obscure inputs.  All other cases where Endpoints are set, there is no possibility of an endpoint being nil

File: operator/pkg/validate/validate.go (hunk 1 above)
Comment (GregHanson) [reply operator/pkg/validate/validate.go:225]: no, I don't think it is valid.  Switching to return an error

**Review by hzxuzhonghu — COMMENTED** [review]
It deserves a test

**Pull request merged**